includes/admin/upgrades/upgrade-functions.php

```diff
@@ -1328,498 +1328,227 @@ function edd_include_file_download_log_migration_batch_processor( $class = '' )
-				<span class="step-instructions-wrapper">
-					<?php wp_nonce_field( 'edd_ajax_export', 'edd_ajax_export' ); ?>
-
-					<?php if ( ! $customer_notes_removal_complete ) : ?>
-						<span class="edd-migration allowed" style="<?php echo $migration_complete ? '' : 'display: none'; ?>">
-							<button type="submit" id="migrate-customer-notes-submit" class="button button-primary"><?php esc_attr_e( 'Upgrade Database', 'easy-digital-downloads' ); ?></button>
-						</span>
-
-						<span class="edd-migration unavailable" style="<?php echo ! $migration_complete ? '' : 'display: none'; ?>">
-							<button type="submit" disabled id="migrate-customer-notes-submit" class="button button-secondary"><?php esc_attr_e( 'Upgrade Database', 'easy-digital-downloads' ); ?></button>
-							&mdash;
-								 <span class="message initial"><?php esc_html_e( 'Please complete the previous steps before updating customer notes.', 'easy-digital-downloads' ); ?></span>
-								 <span class="message complete"><?php esc_html_e( 'Data migration complete.', 'easy-digital-downloads' ); ?></span>
-						</span>
-					<?php else: ?>
-						<button type="submit" disabled id="migrate-customer-notes-submit" class="button button-secondary"><?php esc_attr_e( 'Upgrade Database', 'easy-digital-downloads' ); ?></button>
-						&mdash; <?php esc_html_e( 'Legacy data has already been removed, migration is not possible at this time.', 'easy-digital-downloads' ); ?>
-					<?php endif; ?>
-
-					<input type="hidden" name="edd-export-class" value="EDD\Admin\Upgrades\v3\Customer_Notes" />
+	<div id="edd-v3-remove-legacy-data" <?php echo ! $can_remove_legacy_data ? 'class="edd-hidden"' : ''; ?>>
+		<h2><?php esc_html_e( 'Remove Legacy Data', 'easy-digital-downloads' ); ?></h2>
+		<p>
+			<?php echo wp_kses( __( '<strong>Important:</strong> This removes all legacy data from where it was previously stored in custom post types and post meta. This is an optional step that is not reversible. Please back up your database and ensure your store is operational before completing this step.', 'easy-digital-downloads' ), array( 'strong' => array() ) ); ?>
+		</p>
+		<form class="edd-v3-migration" method="POST">
+			<p>
+				<label for="edd-v3-remove-legacy-data-confirmation">
+					<input type="checkbox" id="edd-v3-remove-legacy-data-confirmation" class="edd-v3-migration-confirmation" name="backup_confirmation" value="1">
+					<?php esc_html_e( 'I have confirmed my store is operational and I have a backup of my website data.', 'easy-digital-downloads' ); ?>
+				</label>
+			</p>
+			<input type="hidden" name="_wpnonce" value="<?php echo esc_attr( wp_create_nonce( 'edd_process_v3_upgrade' ) ); ?>">
+			<input type="hidden" name="upgrade_key" value="v30_legacy_data_removed">
+			<button type="submit" class="button button-primary disabled" disabled>
+				<?php esc_html_e( 'Permanently Remove Legacy Data', 'easy-digital-downloads' ); ?>
+			</button>
+			<!-- @todo align better with button -->
```

Comment:
Would it be possible to wrap the button and this span in a container div and then use some flexbox to align them vertically?

Reply:
For sure. Was waiting to do any styling until the CSS webpack PR was merged.